tools/rvn/CommandLineApp.cs

```diff
@@ -51,6 +61,137 @@ public static int Run(string[] args)
+                        Progress = new SetupProgressAndResult(tuple =>
+                        {
+                            if (tuple.Message != null)
+                            {
+                                Console.WriteLine(tuple.Message);
+                            }
+
+                            if (tuple.Exception != null)
+                            {
+                                Console.Error.WriteLine(tuple.Exception.Message);
+                            }
+                        }),
+                        CancellationToken = token
+                    });
+                });
+            });
+        }
+        
+        private static async Task<int> CreateSetupPackage(CreateSetupPackageParameters parameters)
+        {
+            byte[] zipFile;
+            SetupInfo setupInfo;
+
+            if (File.Exists(parameters.SetupInfoPath) == false)
+                return ExitWithError($"{nameof(parameters.SetupInfoPath)} does not exits", parameters.Command);
+
+            try
+            {
+                using (StreamReader file = File.OpenText(parameters.SetupInfoPath))
+                {
+                    JsonSerializer serializer = new();
+                    setupInfo = (SetupInfo)serializer.Deserialize(file, typeof(SetupInfo));
+                }
+            }
+            catch (Exception e)
+            {
+                return ExitWithError($"Failed to deserialize SetupInfo object form this file path {nameof(parameters.SetupInfoPath)}\nError: {e}",parameters.Command);
+            }
+
+            if (string.IsNullOrEmpty(setupInfo?.Domain))
+                return ExitWithError($"{nameof(setupInfo.Domain)} cannot be empty",parameters.Command);
+            
+            parameters.PackageOutputPath ??= setupInfo.Domain;
+            
+            if (string.IsNullOrEmpty(parameters.CertPassword.PassFromEnv) == false)
+            {
+                setupInfo.Password = parameters.CertPassword.PassFromEnv;
+            }
+            else if (string.IsNullOrEmpty(parameters.CertPassword.passFromCmd) == false)
+            {
+                setupInfo.Password = parameters.CertPassword.PassFromEnv;
+            }
+
+            switch (parameters.SetupMode)
+            {
+                case ImportCertificateMode when string.IsNullOrEmpty(parameters.CertificatePath):
+                {
+                    return ExitWithError("--path option must be set", parameters.Command);
+                }
+                case ImportCertificateMode:
+                {
+                    
+                    var certBytes = await File.ReadAllBytesAsync(parameters.CertificatePath, parameters.CancellationToken);
+                    var certBase64 = Convert.ToBase64String(certBytes);
+                    setupInfo.Certificate = certBase64;
+                    zipFile = await LetsEncryptUtils.ImportCertificateSetup(setupInfo, parameters.Progress, parameters.CancellationToken);
```

Comment:
'Bring your own certificate' option doesn't have anything to do with LetsEncrypt, does it? Maybe it should reside in another class or the class should have a different name.

Reply:
done